lib/global-styles.php

```diff
@@ -130,6 +130,14 @@ function_exists( 'gutenberg_is_edit_site_page' ) &&
 			'__experimentalNoWrapper' => true,
 		);
 
+		// Reset existing global styles.
+		foreach ( $settings['styles'] as $key => $style ) {
+			if ( isset( $style['__unstableType'] ) && 'globalStyles' === $style['__unstableType'] ) {
+				unset( $settings['styles'][ $key ] );
```

Comment:
The problem with "unset" is that the "styles"  will become an object instead of an array when passing to JS, which could have some unintended side effects. I'm fixing that here https://github.com/WordPress/gutenberg/pull/34439/commits/eea97994cd2a9ab7c763e905f02103535ec2daed